src/app/domain/authentication/services/AuthService.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'events'
+
+import Minilog from '@cozy/minilog'
+
+import type CozyClient from 'cozy-client'
+
+import { asyncLogout } from '/libs/intents/localMethods'
+import { safePromise } from '/utils/safePromise'
+
+export const authLogger = Minilog('[AuthService]')
+const emitter = new EventEmitter()
+
+let userRevoked = false
+let clientInstance: CozyClient | null = null
+
+// Setter
+export const setUserRevoked = (status: boolean): void => {
```

Comment:
Same here, `setUserRevoked` goal is to react to local state, but it seems like we can just react to `handleTokenError` and `userConfirmation` calls. My comprehension is that this service would be easier to understand if stateless without any feature loss. I'm I right?

Reply:
Yeah, I guess it can expose only one or two of these methods depending on how the modal display/hide is handled. Overall I think there is room to simplify the feature and remove almost all stateful logic